betterconf/providers.py

```diff
@@ -47,8 +49,8 @@ def get(self, name: str) -> str:
 class TOMLProvider(AbstractProvider):
     """Provider that gets values from .TOML files"""
 
-    def __init__(self, files: Union[str, List[str]]):
-        self.data = toml.load(files)
+    def __init__(self, load: TOMLLoadFunction, files: Union[str, List[str]]):
```

Comment:
I don't actually know how to declare those features.
In Rust, we do `#[cfg(feature="somefeature")]`, how's it done in Python?

Reply:
It could look like the _great_ idea but I think it's just unnecessary over complicating
I believe it would be better to place `TOMLProvider` (and `toml` dependency respectively) to a `toml` feature, so user will be able to use it without odd headache and thinking "what function do i need to pass?"

Reply:
Check out the poetry's documentation